src/main/java/org/sagebionetworks/bridge/spring/controllers/PassthroughController.java

```diff
@@ -0,0 +1,144 @@
+import java.util.concurrent.TimeUnit;
+import javax.servlet.http.HttpServletRequest;
+
+import com.google.common.base.Charsets;
+import com.google.common.base.Stopwatch;
+import org.apache.http.Header;
+import org.apache.http.HttpResponse;
+import org.apache.http.client.fluent.Request;
+import org.apache.http.entity.ContentType;
+import org.apache.http.util.EntityUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.CrossOrigin;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+import org.sagebionetworks.bridge.config.Config;
+
+/**
+ * Pass-through controller. Takes in HTTP requests that aren't caught by any other controller and forwards them to
+ * BridgePF.
+ */
+@CrossOrigin
+@RestController
+public class PassthroughController {
+    private static final Logger LOG = LoggerFactory.getLogger(PassthroughController.class);
+
+    static final String CONFIG_KEY_BRIDGE_PF_HOST = "bridge.pf.host";
+    static final String HEADER_REQUEST_ID = "X-Request-Id";
+
+    private String bridgePfHost;
+
+    /** Bridge config. */
+    @Autowired
+    public void setConfig(Config config) {
+        bridgePfHost = config.get(CONFIG_KEY_BRIDGE_PF_HOST);
+    }
+
+    /** Passthrough handler. */
+    @RequestMapping
+    public ResponseEntity<String> handleDefault(HttpServletRequest request, @RequestBody(required = false) String body)
+            throws IOException {
+        Stopwatch stopwatch = Stopwatch.createStarted();
+
+        // URL. This includes query parameters. Spring provides them to use as a string, so we just append them to the
+        // url like a string.
+        String url = request.getRequestURI();
+        LOG.info("Received request " + request.getMethod() + " " + url);
+
+        String fullUrl = bridgePfHost + url;
+        if (request.getQueryString() != null) {
+            fullUrl += "?" + request.getQueryString();
+        }
+
+        // Method. Note that Apache HTTP has different function calls for each HTTP method, so we need this switch.
+        Request pfRequest;
+        switch (request.getMethod()) {
+            case "GET":
+                pfRequest = Request.Get(fullUrl);
+                break;
+            case "POST":
+                pfRequest = Request.Post(fullUrl);
+                break;
+            case "DELETE":
+                pfRequest = Request.Delete(fullUrl);
+                break;
+            default:
+                String errorMessage = "Method " + request.getMethod() + " not supported";
+                LOG.warn(errorMessage);
+                return ResponseEntity.badRequest().build();
+        }
```

Comment:
We need to pass through the OPTIONS HTTP method, I believe. If you were to access this server through the Bridge study manager, I believe you'd see errors around the failure to process OPTIONS requests. I should try and get this running locally and work on the CORS stuff, I'll do that after you merge.

Reply:
Spring MVC has special handling of the OPTIONS method, so this handler never sees that at all.